applets/clock/clock-location.c

```diff
@@ -210,17 +210,14 @@ clock_location_finalize (GObject *g_obj)
                                               G_CALLBACK (network_changed),
                                               CLOCK_LOCATION (g_obj));
 
-        g_free (priv->name);
-        g_free (priv->city);
+        g_clear_pointer (&priv->name, g_free);
+        g_clear_pointer (&priv->city, g_free);
```

Comment:
What is the advantage of using `g_clear_pointer()` in this case? I know that the pointers are set to `NULL`, but do we need this here?

Reply:
avoid issues with inheritance:

> instance finalization function, should finish the finalization of the instance begun in dispose and chain up to the finalize method of the parent class.

Reply:
Before freeing a variable you can check wether it's NULL, and free it when it isn't NULL, next you can assign the variable to NULL to avoid to free it again, either because it can call dispose several times or because it can call dispose before calling finalize.
```C
/**
 * g_clear_pointer: (skip)
 * @pp: (not nullable): a pointer to a variable, struct member etc. holding a
 *    pointer
 * @destroy: a function to which a gpointer can be passed, to destroy *@pp
 *
 * Clears a reference to a variable.
 *
 * @pp must not be %NULL.
 * 
 * If the reference is %NULL then this function does nothing.
 * Otherwise, the variable is destroyed using @destroy and the
 * pointer is set to %NULL.
 *
 * A macro is also included that allows this function to be used without
 * pointer casts. This will mask any warnings about incompatible function types
 * or calling conventions, so you must ensure that your @destroy function is
 * compatible with being called as `GDestroyNotify` using the standard calling
 * convention for the platform that GLib was compiled for; otherwise the program
 * will experience undefined behaviour.
 *
 * Since: 2.34
 **/
#undef g_clear_pointer
void
g_clear_pointer (gpointer      *pp,
                 GDestroyNotify destroy)
{
  gpointer _p;

  _p = *pp;
  if (_p)
    {
      *pp = NULL;
      destroy (_p);
    }
}
```

Reply:
got it, thanks for the clarification